apollo-federation/src/query_graph/build_query_graph.rs

```diff
@@ -1557,36 +1564,41 @@ impl FederatedQueryGraphBuilder {
                     .or_default()
                     .push(object_field_arg.clone());
                 let field_coordinate = object_field_arg.parent();
-                if let Some(dir) = input_value.directives.get(
```

Comment:
Bugs:
- If the directive isn't present here, we should `bail!()` instead of silently ignoring it.